File: platform/polyglot/figma_api/mjs/src/routes/variables.mjs
/**
 * Variables Routes — Figma API SDK (Enterprise)
 */

export default async function variablesRoutes(server, { variablesClient }) {
  server.get('/files/:fileKey/variables/local', async (request) => {
    return variablesClient.getLocalVariables(request.params.fileKey);
  });

  server.get('/files/:fileKey/variables/published', async (request) => {
    return variablesClient.getPublishedVariables(request.params.fileKey);
  });

  server.post('/files/:fileKey/variables', async (request, reply) => {
    const result = await variablesClient.createVariables(request.params.fileKey, request.body);
    return reply.status(201).send(result);
  });

  return Promise.resolve();
}

File: platform/polyglot/figma_api/py/figma_api/routes/variables.py
"""Variables Routes — Figma API SDK (Enterprise)"""

from typing import Any, Dict

from fastapi import APIRouter, Depends, Request

router = APIRouter()


def get_variables_client(request: Request):
    return request.app.state.variables_client


@router.get("/files/{file_key}/variables/local")
async def get_local_variables(file_key: str, client=Depends(get_variables_client)):
    return await client.get_local_variables(file_key)


@router.get("/files/{file_key}/variables/published")
async def get_published_variables(file_key: str, client=Depends(get_variables_client)):
    return await client.get_published_variables(file_key)


@router.post("/files/{file_key}/variables", status_code=201)
async def create_variables(
    file_key: str, body: Dict[str, Any], client=Depends(get_variables_client)
):
    return await client.create_variables(file_key, body)
